src/persist-txn/src/txn_cache.rs

```diff
@@ -1411,7 +1444,7 @@ mod tests {
         let mut cache = TxnsCache::<_, TxnsCodecDefault>::from_read(txns_read, None).await;
         cache.update_gt(&15).await;
         let snap = cache.data_snapshot(d0, 12);
-        assert_eq!(snap.latest_write, None);
+        assert_eq!(snap.latest_write, Some(5));
```

Comment:
`Some(5)` here is also correct just maybe a bit more pessimistic. The problem is that progress of the cache has not been updated up to `20`, so it has no way of knowing that the write at `5` was applied. In fact I think this test was previously relying on the bug that we would compact writes before knowing that they were applied. I also think this test was likely written before we started encoding the original timestamps in the entries. Still, it's probably worth a close look.

Reply:
Ah yeah, makes sense. There's probably some optimization we could do with the `init_ts` but I'm just as happy as you are to avoid the complexity and leave this as slightly pessimistic.